test/io/ferguson/read-class-reorder.chpl

```diff
@@ -0,0 +1,29 @@
+/*class Base {
+}
+
+class Parent : Base {
+  var x: int;
+  var y: int;
+}
+*/
+
+class Child /*: Parent*/ {
```

Comment:
If I understand what you're describing - it doesn't actually work. Although we can now handle fields being read in a different order than declaration order, if there is inheritance, the parent class fields must be read before the child class fields.